src/_guides/libraries/java-interop.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Java interop using package:jnigen"
+description: "To use Java in your Dart program, use package:jnigen."
+jnigen: "https://pub.dev/packages/jnigen"
+jni: "https://pub.dev/packages/jni"
+example: "https://github.com/HosseinYousefi/jnigen_example/tree/main"
+jnidoc: "https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/jniTOC.html"
```

Comment:
Consider linking to the version of this for the latest JDK LTS, at least if jnigen supports it:

```suggestion
jnidoc: "https://docs.oracle.com/en/java/javase/17/docs/specs/jni/index.html"
```